src/main/kotlin/com/cognifide/gradle/aem/bundle/tasks/BundleCompose.kt

```diff
@@ -1,40 +1,42 @@
-package com.cognifide.gradle.aem.bundle
+package com.cognifide.gradle.aem.bundle.tasks
 
 import aQute.bnd.gradle.BundleTaskConvention
 import com.cognifide.gradle.aem.AemException
 import com.cognifide.gradle.aem.AemExtension
+import com.cognifide.gradle.aem.AemTask
+import com.cognifide.gradle.aem.bundle.BundleException
 import com.cognifide.gradle.aem.common.build.DependencyOptions
 import com.cognifide.gradle.aem.common.instance.service.osgi.Bundle
 import com.cognifide.gradle.aem.common.utils.Formats
 import com.fasterxml.jackson.annotation.JsonIgnore
-import java.io.File
-import java.io.Serializable
 import org.apache.commons.lang3.StringUtils
 import org.apache.commons.lang3.exception.ExceptionUtils
 import org.gradle.api.plugins.JavaPlugin
-import org.gradle.api.tasks.Input
-import org.gradle.api.tasks.Internal
-import org.gradle.api.tasks.Optional
+import org.gradle.api.plugins.JavaPluginConvention
+import org.gradle.api.tasks.*
 import org.gradle.api.tasks.bundling.Jar
+import java.io.File
 
-/**
- * The main purpose of this extension point is to provide a place for specifying custom
- * OSGi bundle related properties, because it is not possible to add properties to existing tasks
- * like 'jar' directly.
- */
 @Suppress("LargeClass", "TooManyFunctions")
-class BundleJar(
-    @Transient
-@JsonIgnore
-private val aem: AemExtension,
+open class BundleCompose : Jar(), AemTask {
+
+    @Internal
+    final override val aem = AemExtension.of(project)
 
     @Internal
-@Transient
-@JsonIgnore
-val jar: Jar
-) : Serializable {
+    val bundleConvention = BundleTaskConvention(this).also { convention.plugins.set("bundle", it) }
 
-    var name = jar.name
+    @Internal
+    val javaConvention = project.convention.getPlugin(JavaPluginConvention::class.java)
+
+    /**
+     * Allows to configure BND tool specific options.
+     *
+     * @see <https://bnd.bndtools.org>
+     */
+    fun bndTool(options: BundleTaskConvention.() -> Unit) {
```

Comment:
and instead of having own `bndFile` , `bndInstructions` properties in GAP, now I am delegating them to BND tool class directly / less to maintain